app/src/main/java/chat/rocket/android/createChannel/ui/CreateNewChannelActivity.kt

```diff
@@ -0,0 +1,187 @@
+package chat.rocket.android.createChannel.ui
+
+import android.app.Activity
+import android.content.Intent
+import android.graphics.drawable.GradientDrawable
+import android.os.Bundle
+import android.support.design.chip.Chip
+import android.support.v7.app.AppCompatActivity
+import android.view.MenuItem
+import chat.rocket.android.R
+import chat.rocket.android.createChannel.addMembers.ui.AddMembersActivity
+import chat.rocket.android.createChannel.presentation.CreateNewChannelPresenter
+import chat.rocket.android.createChannel.presentation.CreateNewChannelView
+import chat.rocket.android.util.extensions.setVisible
+import chat.rocket.android.util.extensions.showToast
+import chat.rocket.common.model.roomTypeOf
+import com.jakewharton.rxbinding2.widget.RxTextView
+import dagger.android.AndroidInjection
+import kotlinx.android.synthetic.main.activity_create_new_channel.*
+import kotlinx.android.synthetic.main.layout_toolbar.*
+import javax.inject.Inject
+
+internal const val ADD_MEMBERS_ACTIVITY_REQUEST_CODE = 1
+
+class CreateNewChannelActivity : AppCompatActivity(), CreateNewChannelView {
+    @Inject
+    lateinit var presenter: CreateNewChannelPresenter
+    private var channelType: String = "public"
```

Comment:
Looking at this class I can see that there are `"public", "c" and "p"` ... wait! `public` does not comply with our API! Also, use `lateinit` here and why not specifying the room type using the proper model instead of String?

Reply:
Yeah, thanks for pointing this out.